### PR TITLE
Do not translate arbitrary precision operations without corresponding extensions

### PR DESCRIPTION
**Pull request by vmaksimo**
This patch addresses https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/653#discussion_r463629986

If the appropriate extension for arbitrary precision instruction is not enabled, invalid IR will be produced. Report an error in this case.

### CODE REVIEW

**Comment by mlychkov** [comment]
@AlexeySotkin this fix addresses your comment https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/653#discussion_r463629986. Could you please review it?

**Comment by mlychkov** [comment]
Tagging @AlexeySachkov @MrSidims for review.

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -2735,17 +2735,25 @@ SPIRVInstruction *LLVMToSPIRV::transBuiltinToInst(StringRef DemangledName,
...
-          ExtensionID::SPV_INTEL_arbitrary_precision_floating_point))
-    return nullptr;
+  if (OpFixedSqrtINTEL <= OC && OC <= OpFixedExpINTEL)
+    BM->getErrorLog().checkError(
+        BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_arbitrary_precision_fixed_point),
+        SPIRVEC_InvalidInstruction,
+        CI->getCalledOperand()->getName().str() +
+            "\nFixed point instructions can't be translated correctly without "
+            "enabled extension!\n");
```
Comment (MrSidims) [inline lib/SPIRV/SPIRVWriter.cpp:2745]: Please add the extension name

**Review by MrSidims — APPROVED** [review]

**Review by MrSidims — COMMENTED** [review]
I think a test shall be added in test/negative directory.

File: test/transcoding/capability-arbitrary-precision-fixed-point-numbers.ll [hunk 2]
```diff
@@ -95,7 +95,7 @@
 ; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_arbitrary_precision_integers,+SPV_INTEL_arbitrary_precision_fixed_point -o %t.spv
 ; RUN: llvm-spirv %t.spv -to-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
 
-; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_arbitrary_precision_integers -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV-NEGATIVE
+; RUN: not llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_arbitrary_precision_integers -spirv-text -o /dev/null
```
Comment (AlexeySachkov) [inline test/transcoding/capability-arbitrary-precision-fixed-point-numbers.ll:98]: Please add a check for error message

**Review by AlexeySachkov — APPROVED** [review]

**Pull request merged**